src/components/ArticleVoter.jsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import { patchArticleVotes } from "../utils";
+
+const ArticleVoter = ({ votes, article_id }) => {
+  const [voteIncrement, setVotes] = useState(0);
+  const [err, setErr] = useState(null);
+
+  const handleVoteIncrease = () => {
+    setVotes(currentVotes => currentVotes + 1);
+    setErr(null);
+    patchArticleVotes(article_id, 1)
+      .then(data => {
+        console.log(data);
+      })
+      .catch(err => {
+        setVotes(currentVotes => currentVotes - 1);
+        setErr("Something went wrong, please try again.");
+      });
+  };
+
+  const handleVoteDecrease = () => {
+    setVotes(currentVotes => currentVotes - 1);
+    setErr(null);
+    patchArticleVotes(article_id, -1)
+      .then(data => {
+        console.log(data);
+      })
+      .catch(err => {
+        setVotes(currentVotes => currentVotes + 1);
+        setErr("Something went wrong, please try again.");
+      });
```

Comment:
we could make this code slightly more `DRY` by having these be one function with some conditional logic going on inside them, nothing to change now just soemthing to consider maybe refactoring in the future!